src/components/Todo.js

```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react';
+import { nanoid } from 'nanoid';
+import { HiPlusCircle } from 'react-icons/hi';
+import TodoList from './TodoList';
+
+const Todo = () => {
+  const [data, setDate] = useState([]);
+  const [input, setInput] = useState('');
+
+  const handleChange = event => {
+    setInput(event.target.value);
+  };
+
+  const handleAdd = () => {
+    const newData = {
+      id: nanoid(),
+      title: input,
+      isChecked: false,
+    };
+    if (input === '') return;
+    setDate(prevData => [...prevData, newData]);
+    setInput('');
+  };
+
+  const handleCompleted = id => {
+    const completed = data.map(data => {
+      return data.id === id ? { ...data, isChecked: !data.isChecked } : data;
+    });
+    setDate(completed);
+  };
+
+  const handleDelete = id => {
+    const deleteData = data.filter(data => {
+      return data.id !== id;
+    });
+    setDate(deleteData);
+  };
+
+  const todoElement = data.map(todo => <TodoList key={todo.id} {...todo} handleDelete={() => handleDelete(todo.id)} handleComplete={() => handleCompleted(todo.id)} />);
```

Comment:
I think it's best to rename the TodoList component to a more appropriate name. It isn't apparent that it is a to-do item based on its name.